File: ios/ios-reference/App/App/Plugins/GameAudioSessionPlugin.swift
import Foundation
import Capacitor

@objc(GameAudioSessionPlugin)
public final class GameAudioSessionPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "GameAudioSessionPlugin"
    public let jsName = "GameAudioSession"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getBridgeInfo", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "configure", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "activate", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "deactivate", returnType: CAPPluginReturnPromise),
    ]

    @objc public func getBridgeInfo(_ call: CAPPluginCall) {
        call.resolve([
            "plugin": "GameAudioSession",
            "bridgeVersion": 1,
            "platform": "ios",
            "availability": "available",
            "implementation": "skeleton",
            "capabilities": ["audio_focus", "interruption_events", "silent_mode_policy"]
        ])
    }

    @objc public func configure(_ call: CAPPluginCall) {
        rejectNotImplemented(call, method: "configure")
    }

    @objc public func activate(_ call: CAPPluginCall) {
        rejectNotImplemented(call, method: "activate")
    }

    @objc public func deactivate(_ call: CAPPluginCall) {
        rejectNotImplemented(call, method: "deactivate")
    }

    private func rejectNotImplemented(_ call: CAPPluginCall, method: String) {
        call.reject("GameAudioSession.\(method) is not implemented in Phase 04.", "NOT_IMPLEMENTED")
    }
}

File: android-reference/app/src/main/java/com/ceegore/riftwarden/plugins/GameAudioSessionPlugin.java
package com.ceegore.riftwarden.plugins;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "GameAudioSession")
public final class GameAudioSessionPlugin extends Plugin {
    private static final int BRIDGE_VERSION = 1;

    @PluginMethod
    public void getBridgeInfo(PluginCall call) {
        JSObject result = new JSObject();
        result.put("plugin", "GameAudioSession");
        result.put("bridgeVersion", BRIDGE_VERSION);
        result.put("platform", "android");
        result.put("availability", "available");
        result.put("implementation", "skeleton");
        result.put("capabilities", new String[] { "audio_focus", "interruption_events", "silent_mode_policy" });
        call.resolve(result);
    }

    @PluginMethod
    public void configure(PluginCall call) { rejectNotImplemented(call, "configure"); }

    @PluginMethod
    public void activate(PluginCall call) { rejectNotImplemented(call, "activate"); }

    @PluginMethod
    public void deactivate(PluginCall call) { rejectNotImplemented(call, "deactivate"); }

    private void rejectNotImplemented(PluginCall call, String method) {
        call.reject("GameAudioSession." + method + " is not implemented in Phase 04.", "NOT_IMPLEMENTED");
    }
}
